content/faq/multiple-servers.md

```diff
@@ -39,8 +39,41 @@ const app = await NestFactory.create(
 );
 await app.init();
 
-http.createServer(server).listen(3000);
-https.createServer(httpsOptions, server).listen(443);
+const httpServer = http.createServer(server).listen(3000);
+const httpsServer = https.createServer(httpsOptions, server).listen(443);
+```
+
+Because we called `http.createServer` / `https.createServer` ourselves, NestJS doesn't close them when calling `app.close` / on termination signal. We need to do this ourselves:
+
+```typescript
+@Injectable()
+export class ShutdownObserver implements OnApplicationShutdown {
```

Comment:
can we clarify somehow where to register this `ShutdowObserver`? maybe in `AppModule`?

Reply:
Like how? Something like

```
Add `ShutdownObserver` to the `providers` of the `AppModule`
```

?

Reply:
yeah, sounds good